packages/scratch-vm/src/extensions/scratch3_musiccreation/index.js

```diff
@@ -568,14 +589,24 @@ class Scratch3MusicCreation {
      */
     async _getAndPlayMagentaNotes(RNN, args, utils, inst) {
         let magenta_notes = null;
+        let valid = true;
         if (RNN) {
-            magenta_notes = await this.musicAccompanimentHelper.testMagentaRNN(this.noteList, args, utils);
+            if (this.noteList.length > 0) {
```

Comment:
Since there's so much nesting here, it might be good to split this up into separate functions.

Something I like from c# is using a `TryGet` kind of pattern: https://zogface.blog/2017/08/03/using-the-tryget-pattern-in-c-to-clean-up-your-code/

which might work for you here. in js you have to approach it a little differently, since you don't have any `out` keyword. How I might do this is the following:

```js

// Current function
async _getAndPlayMagentaNotes(RNN, args, utils, inst) {
  const result = tryGetMagentaNotes(RNN, args, utils, inst);

  if (!result.valid) {
    utils.stackFrame.duration = 0;
    return;
  }

  const prepared_notes = this._prepare(result.notes);
  this.magentaNoteList = prepared_notes['notes'];
  this.musicCreationHelper.playNotes(prepared_notes, utils, inst,this.vizHelper); 
}

// New function
async tryGetMagentaNotes(RNN, args, utils, inst) {
  if (!RNN) {
    const notes = await this.musicAccompanimentHelper.testMagentaMVAE(utils);
    return { notes, valid: true };
  }

  ...
  // Handle logic for get the RNN notes.  
  // Should return either {notes, valid: true} or {notes, valid: false} 
}
```